app/experimenter/visualization/api/v3/views.py

```diff
@@ -110,9 +110,30 @@ def append_conversion_count(results, primary_metrics_set):
             ] = conversion_count
 
 
-def process_data_for_consumption(data, experiment):
-    append_population_percentages(data)
-    results, primary_metrics_set = generate_results_object(data, experiment)
+def get_week_x_retention(week_index, weekly_data):
+    return list(
+        filter(
+            lambda row: row["window_index"] == str(week_index)
+            and row["metric"] == "retained",
+            weekly_data or [],
+        )
+    )
```

Comment:
Generally list comprehensions are preferred over filter/lambda but it's not a huge deal